### DISCUSSION TITLE
Renders on useMutation

### DISCUSSION THREAD

**Discussion by reinisg** [TanStack/query#5283] [Q&A] [answered]
Is it possible to avoid from multiple renders?
I have example [codesandbox](https://codesandbox.io/s/dreamy-rhodes-l8kq1u) if we are pressing first button "Button with Mutate", can see that it is rendering when button is clicked and after useMutation. Is there way to get only one, after mutation? 

Using version 3.39.3.
Hope for some tips to this case.

**Comment by TkDodo** [comment, marked as answer]
no. your component re-renders because `useMutation` gives you meta information about the mutation, like `loading` state. `useMutation` doesn't have the optimizations that `useQuery` has where it can detect and track what you are actually using and what not ...

**Reply by reinisg** [reply]
Thanks for the fast reply and description!
Okay, then I will try to fix in another way!
